plugins/pipelines/app/models/concerns/samson_pipelines/stage_concern.rb

```diff
@@ -1,13 +1,28 @@
 module SamsonPipelines::StageConcern
-  # Return true if any stages in the pipeline are marked production
-  def production?
-    super || next_stages.any?(&:production?)
-  end
 
   def next_stages
     Stage.find(next_stage_ids)
   end
 
+  # Needs to find all the possible stages in case this is a pipeline of pipelines as each subsequent stage
+  # could have valid next_stage_ids
+  def all_next_stages
+    Stage.find(recursive_next_stage_ids);
+  end
+
+  def recursive_next_stage_ids
+    return [] if next_stage_ids.empty?
+    next_stage_ids.map(&:to_i) + Stage.find(next_stage_ids).flat_map(&:recursive_next_stage_ids)
+  end
+
+  def deploy_requires_approval?
+    super || all_next_stages.any? do |next_stage|
+      next_stage.deploy_requires_approval? && !next_stage.no_code_deployed?
+    end
+  end
+
+  private :all_next_stages
```

Comment:
seems to cause issues with SseRailsEngine.send_event() when these are all made private (specifically next_stages())